frontend/src/controllers/ChannelCodesController.es6

```diff
@@ -47,7 +47,19 @@ export default class {
     }
 
     populateChannels(channels) {
-        this.$scope.channels = Object.keys(channels).map(k => { return {"name": k, "codes": channels[k]} })
+        this.$scope.channels = Object.keys(channels).map((k, i) => { 
+            const promoCodes = channels[k];
+
+            if (this.$scope.regenerateCodes) {
+                const regeneratedPromoCodes = promoCodes.map((_, ii) => `${this.generateSuggestedPromoCode()}${i}${ii}`);
```

Comment:
I just added a commit to append the index `i` of `Object.keys(channels).map` and he index `ii` of `promoCodes.map` to the  return value of `this.generateSuggestedPromoCode()`.

I made this change as `this.generateSuggestedPromoCode()` was returning the same value when called multiple times as part of iteration - these duplicates occur as `generateSuggestedPromoCode()` uses `new Date().getTime().toString(36)` when generating a promo code, however when called successively that computation returns the same value as `Date().getTime()` isn't granular enough to be unique, I tried drilling down to the smallest time fraction (milliseconds) but it still wasn't granular enough.

These indexes `i` and `ii` make the `regeneratedPromoCode` unique.

Reply:
can we call these something more descriptive than `i` and `ii`?  e.g. `channelIndex` and `promoCodeIndex`

Reply:
not to mentioned the existing `k` which I guess is the `promoChannelName`

Reply:
yep, it was setup as `k` for "key" I presume, which is the Promo Channel Name. I've updated `i`/`ii` as per your suggestion